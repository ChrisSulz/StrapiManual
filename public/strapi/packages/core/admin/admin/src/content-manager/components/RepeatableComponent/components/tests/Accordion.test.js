import * as React from 'react';

import { lightTheme, ThemeProvider } from '@strapi/design-system';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { IntlProvider } from 'react-intl';

import * as Accordion from '../Accordion';

describe('RepeatableComponent | Accordion', () => {
  describe('Group', () => {
    const defaultProps = {
      children: 'I am a child',
    };

    const TestComponent = (props) => (
      <ThemeProvider theme={lightTheme}>
        <IntlProvider locale="en" messages={{}} defaultLocale="en">
          <Accordion.Group {...defaultProps} {...props} />
        </IntlProvider>
      </ThemeProvider>
    );

    const setup = (props) => render(<TestComponent {...props} />);

    it('should render the children passed to it', () => {
      const { container, getByText } = setup();

      expect(getByText('I am a child')).toBeInTheDocument();

      expect(container).toMatchSnapshot();
    });

    it('should render the error if there is one', () => {
      const { container, getByText } = setup({
        error: { id: 'error', defaultMessage: 'I have an error' },
      });

      expect(getByText('I have an error')).toBeInTheDocument();

      expect(container).toMatchSnapshot();
    });

    it('should make the children keyboard navigable', async () => {
      const user = userEvent.setup();

      const { getByText } = setup({
        children: (
          <div>
            <div>
              <button data-strapi-accordion-toggle type="button">
                I am a first button
              </button>
            </div>
            <div>
              <button data-strapi-accordion-toggle type="button">
                I am a second button
              </button>
            </div>
          </div>
        ),
      });

      await user.tab();

      expect(getByText('I am a first button')).toHaveFocus();

      await user.keyboard('[ArrowDown]');

      expect(getByText('I am a second button')).toHaveFocus();
    });
  });

  describe('Content', () => {
    it('should render and match the snapshot', () => {
      const { container } = render(
        <ThemeProvider theme={lightTheme}>
          <Accordion.Content />
        </ThemeProvider>
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe('Footer', () => {
    it('should render and match the snapshot', () => {
      const { container } = render(
        <ThemeProvider theme={lightTheme}>
          <Accordion.Footer />
        </ThemeProvider>
      );
      expect(container).toMatchSnapshot();
    });
  });
});
