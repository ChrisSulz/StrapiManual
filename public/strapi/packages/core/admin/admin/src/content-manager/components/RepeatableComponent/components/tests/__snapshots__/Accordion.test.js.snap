// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepeatableComponent | Accordion Content should render and match the snapshot 1`] = `
.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  border-bottom: none;
}

.c0 > div > div {
  border: 1px solid #dcdce4;
  border-top-color: transparent;
}

.c0 > div:first-child > div {
  border-top: 1px solid #dcdce4;
}

.c0 > div > div,
.c0 > div > div > div {
  border-radius: unset;
}

.c0 > div:first-child > div,
.c0 > div:first-child > div > div {
  border-radius: 4px 4px 0 0;
}

.c0 > div > div[data-strapi-expanded='true'] {
  border: 1px solid #4945ff;
}

<div>
  <div
    class="c0"
  />
  <div
    class="c1"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`RepeatableComponent | Accordion Footer should render and match the snapshot 1`] = `
.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  overflow: hidden;
  border-bottom: 1px solid #dcdce4;
  border-right: 1px solid #dcdce4;
  border-left: 1px solid #dcdce4;
  border-radius: 0 0 4px 4px;
}

<div>
  <div
    class="c0"
  />
  <div
    class="c1"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`RepeatableComponent | Accordion Group should render the children passed to it 1`] = `
.c0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
  <div
    class=""
  >
    I am a child
  </div>
  <div
    class="c0"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`RepeatableComponent | Accordion Group should render the error if there is one 1`] = `
.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c1 {
  font-size: 0.75rem;
  line-height: 1.33;
  color: #d02b20;
}

.c0 {
  padding-top: 4px;
}

<div>
  <div
    class=""
  >
    I am a child
    <div
      class="c0"
    >
      <span
        class="c1"
      >
        I have an error
      </span>
    </div>
  </div>
  <div
    class="c2"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
